runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -203,6 +215,48 @@ private boolean checkCoreArchitecture() {
         return true;
     }
 
+    private boolean verifyPackageSignature(String packageName, String fingerprint) {
+        Signature[] signatures = null;
+        try {
+            PackageManager packageManager = XWalkApplication.getApplication().getPackageManager();
+            PackageInfo packageInfo = packageManager.getPackageInfo(
+                    packageName, PackageManager.GET_SIGNATURES);
+            signatures = packageInfo.signatures;
+        } catch (NameNotFoundException e) {
+            Log.d(TAG, packageName + " not found");
+            return false;
+        }
+
+        if (signatures == null) {
+            Log.d(TAG, "No signature in package");
+            return false;
+        }
+
+        for (int i = 0; i < signatures.length; ++i) {
+            Log.d(TAG, "Checking signature " + i);
+            byte[] signature = signatures[i].toByteArray();
+
+            try {
+                MessageDigest md = MessageDigest.getInstance("SHA1");
```

Comment:
No, we can use sha256.  You probably want to check the edss config (the stuff on codesign.intel.com), because Matt tells me that the default hashing algorithm in there is currently SHA1 (it used to be 256 but was reverted due to some compatibility) and you'll need to make sure it's set explicitly to sha256 (and not set to sha1 or left blank).